pkg/expansion/expand_test.go

```diff
@@ -101,11 +106,13 @@ func TestMappingDual(t *testing.T) {
 	doExpansionTest(t, mapping)
 }
 
-func doExpansionTest(t *testing.T, mapping func(string) string) {
+func doExpansionTest(t *testing.T, mapping func(string) (string, error)) {
+	unknownVarErrorType := reflect.TypeOf(&UnknownVarError{})
```

Comment:
is there a less gross way of doing this?

Reply:
I suggest following the io.EOF in https://golang.org/pkg/io/#pkg-variables

var UNKNOWN_VAR = errors.New("Unknown variable")  
Then just check for err == UNKNOWN_VAR.

Reply:
This was my first thought, but I wanted to parameterize the string representation. See: https://github.com/kubernetes-sigs/kustomize/pull/442/files#diff-b91a8276fd8165b74c5dbdac88df5cc0R42

Reply:
Can you pencil in your overall design for this PR in the initial comment?

Reply:
Updated the comment.